git/remote.py

```diff
@@ -990,6 +1028,16 @@ def pull(
             # No argument refspec, then ensure the repo's config has a fetch refspec.
             self._assert_refspec()
         kwargs = add_progress(kwargs, self.repo.git, progress)
+
+        if not allow_unsafe_protocols and refspec:
+            if isinstance(refspec, str):
```

Comment:
I think the `Git` type might already have a function to flatten strings to lists, maybe something like it can be factored into its own function to allow something like:

```python
          if not allow_unsafe_protocols and refspec:
                 for ref in to_str_list(refspec):
                     Git.check_unsafe_protocols(ref)
```